main.tf

```diff
@@ -149,14 +168,76 @@ resource "kubernetes_deployment" "wandb" {
               memory = "8G"
             }
           }
+
+          args = [ "/bin/sh", "-c", "mkdir -p /var/log/nginx" ]
```

Comment:
This isn't going to work.  If you need to make sure a directory exists on the mounted volume you should use an [init container](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).  It's also going to be much easier to test this with a local k8s cluster like [kind](https://kind.sigs.k8s.io/) and modifying a local [deployment.yaml](https://docs.wandb.ai/guides/self-hosted/setup/on-premise-baremetal#kubernetes-deployment).

One trick is going to be file permissions.  The container runs as user "wandb" which is UID 999, all processes will also by a member of the root group GID 0.  All files and folders in this volume must be writeable by GID 0 as the UID can change in certain environments like OpenShift.

You can see an example of ensuring permissions for a mounted volume from an init container in our [official helm chart](https://github.com/wandb/helm-charts/blob/main/charts/wandb/templates/mysql.yaml#L37-L51).

Reply:
Hey Chris, thanks for taking a look! I was unaware of the init container concept but this looks like what I was trying to accomplish earlier.

I ended up adding a volume mount with /var/log/nginx to the wandb container rather than trying to create the directory which accomplished what I was trying to resolve which was to [work around this error](https://stackoverflow.com/questions/51200959/docker-compose-volume-behaviour)

Reply:
Ah that only gives me access to the nginx logs from the sidecars then